src/main/java/use_cases/turn_start_use_case/TurnStartInteractor.java

```diff
@@ -0,0 +1,198 @@
+package use_cases.turn_start_use_case;
+
+import entities.GameState;
+import entities.Player;
+import entities.cardEffects.PlayerStatsEffect;
+import entities.cards.Card;
+import entities.cards.CreatureCard;
+import entities.cards.StructureCard;
+import entities.decks.EssenceDeck;
+import entities.decks.PlayerDeck;
+import entities.cardEffects.CreatureStatsEffect;
+import entities.cardEffects.CardEffect;
+import entities.GameEvent;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * TurnStartInteractor implements the methods needed for the logic of
+ * start-of-turn phase.
+ */
+public class TurnStartInteractor implements TurnStartInputBoundary {
+
+    final GameState gameState;
+
+    final TurnStartOutputBoundary turnStartPresenter;
+
+    public TurnStartInteractor(GameState gameState, TurnStartOutputBoundary turnStartPresenter) {
+        this.gameState = gameState;
+        this.turnStartPresenter = turnStartPresenter;
+    }
+
+    /**
+     * Draws cards for the current player: 2 cards from their play deck and 2
+     * more cards from their Essence deck (4 in total).
+     *
+     * @return a DrawCardOutputModel, showing which cards are successfully added
+     * to the hand and which are discarded due to hand being full.
+     */
+    @Override
+    public DrawCardOutputModel drawCards() {
+
+        //Getting the player deck from the current player
+        Player player = gameState.getCurrentPlayer();
+        PlayerDeck deck = player.getPlayerDeck();
+
+        //Getting the essence deck from the current player
+        EssenceDeck edeck = player.getEssenceDeck();
+
+        //Drawing twice from the player deck then from the essence deck
+        boolean val1;
+        boolean val2;
+        List<Integer> burntIds = new ArrayList<>();
+        List<String> burntNames = new ArrayList<>();
+        List<Integer> keptIds = new ArrayList<>();
+        List<String> keptNames = new ArrayList<>();
+        for (int i = 0; i < 2; i++) {
+            Card card = deck.draw();
+            if (card != null) {
+                //going to find kept and discarded cards
+                val1 = player.addCard(card);
+                if (val1) {
+                    keptIds.add(card.getId());
+                    keptNames.add(card.getName());
+                } else {
+                    burntIds.add(card.getId());
+                    burntNames.add(card.getName());
```

Comment:
If a player has the hand capacity of cards, then when they draw the cards, it will automatically be discarded right?

Reply:
yes, `val1` is the boolean value that `addCard()` sends out, it tells us whether adding the card was successful